--- a/util/wanpipemon/fe_lib.h
+++ b/util/wanpipemon/fe_lib.h
@@ -71,6 +71,7 @@ extern int		set_lb_modes_per_chan(u_int8_t, u_int8_t, u_int32_t);
 extern u_int32_t	get_lb_modes(int);
 extern void read_te1_56k_stat(unsigned char);
 extern void read_te1_56k_config (void);
+extern int write_te1_56k_config (void);
 
 extern void set_debug_mode(unsigned char type, unsigned char mode);
 extern void set_fe_debug_mode(sdla_fe_debug_t *);
